lightning/src/ln/channelmanager.rs

```diff
@@ -2841,15 +2857,44 @@ where
 		if let Some(update) = update_opt {
 			// Try to send the `BroadcastChannelUpdate` to the peer we just force-closed on, but if
 			// not try to broadcast it via whatever peer we have.
+			let brodcast_message_evt = events::MessageSendEvent::BroadcastChannelUpdate {
+				msg: update
+			};
+
 			let per_peer_state = self.per_peer_state.read().unwrap();
 			let a_peer_state_opt = per_peer_state.get(peer_node_id)
 				.ok_or(per_peer_state.values().next());
+
+			// if we were able to get the peer we just force closed on.
 			if let Ok(a_peer_state_mutex) = a_peer_state_opt {
 				let mut a_peer_state = a_peer_state_mutex.lock().unwrap();
-				a_peer_state.pending_msg_events.push(events::MessageSendEvent::BroadcastChannelUpdate {
-					msg: update
-				});
+				a_peer_state.pending_msg_events.push(brodcast_message_evt);
+			}
+
+			// if we connected to some other random peer.
+			else if let Err(a_peer_state_mutex) = a_peer_state_opt {
+				match a_peer_state_mutex {
+					// if we were able to connect to Some peer.
+					Some(val) => {
+						let mut a_peer_state = val.lock().unwrap();
+						a_peer_state.pending_msg_events.push(brodcast_message_evt);
+						log_info!(self.logger,
+							"Not able to broadcast channel update to peer we force-closed on. Broadcasting to some random peer.");
```

Comment:
I want to make sure I'm on the right track with your question. Are you suggesting simplifying the conditions to just two possibilities: selecting either Some peer (whether it's the node we disconnected from or another) or None peer? This way, we could streamline the process and potentially eliminate the need for other cases. Let me know if I'm on the right page or if there's more context I am missing here!

Reply:
I have simplified the logic to two cases. See [648a214](https://github.com/lightningdevkit/rust-lightning/commit/648a214968bd904d95939e53fbbbd698d237df74), and https://github.com/lightningdevkit/rust-lightning/pull/2731#issuecomment-1817484363

Let me know if there are further improvements here I can work on!

Reply:
Hi Matt!
Sorry, I missed your latest message on this thread!
Let me try to answer your question!

According to the latest behavior of PR, we only use the pending broadcast queue if we are not connected to any peer.
So we use the event of connecting to a new peer (fn `peer_connected`) as the trigger to clear this broadcast queue and send this msg event through that peer.

If we populate the pending broadcast queue instead of first searching for a random peer to broadcast to, we will delay sending these messages when they can be sent faster.

Reply:
Addressed in [163d535](https://github.com/lightningdevkit/rust-lightning/commit/163d535a72bab5c3bc9608c37fa9fc665989e7d1)

Thanks a lot for the suggestion!